tests/sentry/deletions/test_detector.py

```diff
@@ -0,0 +1,64 @@
+from sentry.deletions.tasks.scheduled import run_scheduled_deletions
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.snuba.models import QuerySubscription
+from sentry.testutils.hybrid_cloud import HybridCloudTestMixin
+from sentry.workflow_engine.models import (
+    DataCondition,
+    DataConditionGroup,
+    DataSource,
+    DataSourceDetector,
+    Detector,
+    DetectorWorkflow,
+)
+from tests.sentry.workflow_engine.test_base import BaseWorkflowTest
+
+
+class DeleteDetectorTest(BaseWorkflowTest, HybridCloudTestMixin):
+    def setUp(self):
+        self.data_condition_group = self.create_data_condition_group()
+        self.data_condition = self.create_data_condition(condition_group=self.data_condition_group)
+        self.snuba_query = self.create_snuba_query()
```

Comment:
If the query subscription is deleted does that mean we always want to delete the related snuba query? We don't have this set up today but I'm not sure why - if I implemented it for this it could affect other things.

Reply:
Yeah, we could do that. In the past we didn't because potentially one snuba query could be shared, but in practice we haven't really used that. 

Could make sense to just check that there aren't other links to the snuba query and then delete it if orphaned.